src/app/services/apiservice.ts

```diff
@@ -1,36 +1,47 @@
 import { Injectable } from "@angular/core";
-import { HttpClient, HttpParams } from "@angular/common/http";
+import { HttpClient, HttpParams, HttpHeaders } from "@angular/common/http";
 import { environment } from "../../environments/environment";
+import { reject, resolve } from "q";
 @Injectable()
 export class ApiService {
+  token: any;
+  role: any;
+  httpOptions = {
+    headers: new HttpHeaders({
+      "Content-Type": "application/json",
+      api_token: localStorage.getItem("token")
+    })
+  };
   constructor(private http: HttpClient) {}
 
   postlogin(post) {
-    const params = new HttpParams()
-      .set("username", post.email)
-      .set("password", post.password);
+    const apidata = { email: post.email, password: post.password };
+
     return new Promise((resolve, reject) => {
       this.http
-        .get(`${environment["apiBase"]}login`, {
-          params: params
-        })
+        .post(`${environment["apiBase"]}login`, apidata)
         .subscribe(data => {
           if (data["error"] && data["error"] === 1) {
             reject(data);
           } else {
             resolve(data);
+            this.token = data["data"].api_token;
```

Comment:
```
this.token = data["data"].api_token;
            localStorage.setItem("role", data["data"].role);
            localStorage.setItem("token", this.token);
```
these function will not work because you resolved function response above them.